montecarlo/run_sim.sh

```diff
@@ -61,11 +72,28 @@ elif [[ ${CONFIG} == *"Rn222"* ]]; then
 fi
  
 # preinit file for Geant4
-PREINIT=preinit.mac
-if [[ ${CONFIG} == *"Cs137"* ]]; then
-    PREINIT=preinit_${CONFIG}.mac
-elif [[ ${CONFIG} == *"muon"* || ${CONFIG} == *"MV"* ]]; then
-    PREINIT=preinit_MV.mac
+PREINIT_MACRO=$8
+if [[ -z $PREINIT_MACRO ]];
+then
+    PREINIT_MACRO=preinit.mac
+    if [[ ${CONFIG} == *"Cs137"* ]]; then
+        PREINIT_MACRO=preinit_${CONFIG}.mac
+    elif [[ ${CONFIG} == *"muon"* || ${CONFIG} == *"MV"* ]]; then
+        PREINIT_MACRO=preinit_MV.mac
+    fi
+fi
+
+OPTICAL_SETUP=$9
+if [[ -z $OPTICAL_SETUP ]];
+then
+    OPTICAL_SETUP=setup_optical_S1.mac
+fi
+
+shift
```

Comment:
pycharm indicated that ${10} was an invalid identifier so I assumed bash only lets you access 9 variables at a time but looking online, it looks like ${10} should work.  I'll update and test it and commit the new changes if it works.